cmd/server/main.go

```diff
@@ -49,6 +51,7 @@ type ConnectorWithCustomTemplate struct {
 	TemplateFile      string `yaml:"template_file"`
 	WebhookURL        string `yaml:"webhook_url"`
 	EscapeUnderscores bool   `yaml:"escape_underscores"`
+	RetryMax          int    `yaml:"retry_max"`
```

Comment:
I would not consider `retry_max` to be a webhook specific configuration but rather to be a global configuration. So, I would suggest to provide `retry-max` as a flag to the cli command. Sounds reasonable?

Reply:
I have halted my review on this line as this will change a lot of what you've changed afterwards.